app/views/announcements/_form.html.erb

```diff
@@ -4,29 +4,52 @@
     bold: true,
   },
   name: "announcement[title]",
+  value: @announcement.title,
   id: "title",
   error_message: error_items(@announcement.errors.messages, :title),
 } %>
+
 <%= render "govuk_publishing_components/components/input", {
   label: {
     text: "Enter the path to the announcement",
     bold: true,
   },
   name: "announcement[path]",
+  value: @announcement.path,
   id: "path",
   error_message: error_items(@announcement.errors.messages, :path),
   hint: "Example format: /government/news/coronavirus-covid-19",
 } %>
+
 <% legend = capture do %>
   <span class="govuk-heading-s govuk-!-margin-bottom-0"><%= "Enter the date of publication" %></span>
 <% end %>
+
 <%= render "govuk_publishing_components/components/date_input", {
   name: "announcement[published_at]",
   id: "published_at",
   legend_text: legend,
   hint: "For example, 31 3 2020",
   error_message: error_items(@announcement.errors.messages, :published_at),
+  items: [
+    {
+      name: "day",
+      width: 2,
+      value: @announcement.published_at&.day
```

Comment:
what will happen if there isn't a published date? does this fall back to nil?

Reply:
Yes, it's doesn't show a value. The `&` is a safety navigation operator and is the equivalent of doing:
```
@announcement.published_at&.day if @announcement.published_at.present? && @announcement.published_at.day.present?
```